spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsStreamListenerSetupMethodOrchestrator.java

```diff
@@ -291,7 +291,10 @@ else if (parameterType.isAssignableFrom(KTable.class)) {
 									Serde<?> keySerde, Serde<?> valueSerde) {
 		KStream<?, ?> stream = streamsBuilder.stream(bindingServiceProperties.getBindingDestination(inboundName),
 				Consumed.with(keySerde, valueSerde));
-		if (bindingProperties.getConsumer().isUseNativeDecoding()){
+		final boolean nativeDecoding = bindingProperties.getConsumer() != null &&
```

Comment:
Won't it be much safer to do this on the Spring Cloud Stream Core level in the `BindingProperties`:
```
private ConsumerProperties consumer = new  ConsumerProperties();

private ProducerProperties producer = new ProducerProperties();
```

At least this is what I do typically for any `@ConfigurationProperties`.

Also that looks strange that this object is `null` according Spring Boot Docs: https://docs.spring.io/spring-boot/docs/2.0.0.RELEASE/reference/htmlsingle/#boot-features-external-config-typesafe-configuration-properties

> If nested POJO properties are initialized (like the `Security` field in the preceding example), a setter is not required. If you want the binder to create the instance on the fly by using its default constructor, you need a setter.

So, or we use this class (`BindingProperties`) directly, then it is wrong and `consumer` and `producer`  have to be initialized on place, or something is wrong with Boot model so far.

On the other hand it is not clear why we have this API there in the `BindingServiceProperties`:
```
public BindingProperties getBindingProperties(String bindingName) {
	BindingProperties bindingProperties = new BindingProperties();
	if (this.bindings.containsKey(bindingName)) {
		BeanUtils.copyProperties(this.bindings.get(bindingName), bindingProperties);
	}
	if (bindingProperties.getDestination() == null) {
		bindingProperties.setDestination(bindingName);
	}
	return bindingProperties;
}
```

I know that there is no panacea from `NPE`, but I feel if we use conventions provided by Boot, we don't need to have such a protection.

I can merge, of course, as is if you are not agree with me (although... arguments to the studio!). Not a big deal. But you should be ready to meet similar problem in other places. 

BTW, why there is no test-case to demonstrate an issue? 
I'm asking because I am surprised to see this problem, when Spring Boot says that it is going to create one for us via its binder if we have setter.

Reply:
Hi @artembilan if the user does not provide a `spring.cloud.stream.bindings.<input>.consumer` property, then boot won't create `ConsumerProperties` for us and that's why the NPE occurs. It's not required in Kafka streams binder and if that's the case, we just get by that.

Reply:
the NPE can be reproduced by removing the conusmer properties from an existing test.

Reply:
OK! So, I think Boot can't create it for us because it is a nested in the `Map`:
```
private Map<String, BindingProperties> bindings = new TreeMap<>(
			String.CASE_INSENSITIVE_ORDER);
```
This is mostly question to Boot then...

Anyway: why don't use `new` then? And from there WDYT about my concern:
> But you should be ready to meet similar problem in other places.

My point is that I would prefer to see a fix in the Core, not here.
That's why I'm so picky.

And you if say that you know how to reproduce why don't cover the fix with the test?